### PR TITLE
Coverage for: trie

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- increase code coverage
  
## Proposed changes
- added unit tests for trie package + some on update package

## Testing procedure
- unit tests only

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by BeniaminDrasovean — APPROVED** [review]

File: trie/factory/trieCreator_test.go [hunk 1]
```diff
@@ -165,13 +165,49 @@ func TestTrieCreator_CreateWithNilCheckpointsStorerShouldErr(t *testing.T) {
...
+
+	args := getArgs()
+	tf, _ := factory.NewTrieFactory(args)
+
+	createArgs := getCreateArgs()
+	createArgs.MaxTrieLevelInMem = 0
+	_, tr, err := tf.Create(createArgs)
+	require.Nil(t, tr)
+	require.NotNil(t, err)
+	require.True(t, strings.Contains(err.Error(), trie.ErrInvalidLevelValue.Error()))
```
Comment (bogdan-rosianu) [inline trie/factory/trieCreator_test.go:179]: could use `require.Contains`

File: trie/snapshotTrieStorageManager_test.go [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
 
 import (
+	"errors"
 	"strings"
 	"testing"
 
 	"github.com/multiversx/mx-chain-core-go/core"
 	"github.com/multiversx/mx-chain-core-go/core/check"
 	"github.com/multiversx/mx-chain-go/common"
+	errorsMx "github.com/multiversx/mx-chain-go/errors"
```
Comment (bogdan-rosianu) [inline trie/snapshotTrieStorageManager_test.go:11]: might rename to `internalErrors`

File: trie/snapshotTrieStorageManager_test.go [hunk 3]
```diff
@@ -31,55 +34,117 @@ func TestNewSnapshotTrieStorageManager(t *testing.T) {
...
+		}
+		stsm, _ := newSnapshotTrieStorageManager(trieStorage, 0)
+
+		_, _ = stsm.Get([]byte("key"))
+		assert.True(t, getFromOldEpochsWithoutCacheCalled)
+	})
 }
 
-func TestNewSnapshotTrieStorageManager_PutWithoutCache(t *testing.T) {
+func TestNewSnapshotTrieStorageManager_Put(t *testing.T) {
```
Comment (bogdan-rosianu) [inline trie/snapshotTrieStorageManager_test.go:85]: Might rename from `TestNewSnapshotTrieStorageManager*` to `TestSnapshotTrieStorageManager*`

Comment (bogdan-rosianu) [reply trie/snapshotTrieStorageManager_test.go:85]: for the entire file

File: trie/trieStorageManagerInEpoch_test.go [hunk 4]
```diff
@@ -1,21 +1,24 @@
 package trie
 
 import (
+	"errors"
 	"strings"
 	"testing"
 
-	"github.com/multiversx/mx-chain-core-go/core/check"
+	errorsMx "github.com/multiversx/mx-chain-go/errors"
```
Comment (bogdan-rosianu) [inline trie/trieStorageManagerInEpoch_test.go:8]: `internalErrors`

File: update/container/accountDBSyncers_test.go [hunk 5]
```diff
@@ -10,89 +10,169 @@ import (
...
+	testAccountsDBSyncersVal1 := &mock.AccountsDBSyncerStub{}
+
+	t.Run("different lengths should error", func(t *testing.T) {
+		t.Parallel()
+
+		adsc := NewAccountsDBSyncersContainer()
+		err := adsc.AddMultiple([]string{testKey0}, nil)
+		require.Equal(t, update.ErrLenMismatch, err)
+	})
+	t.Run("duplicated keys should should error on Add", func(t *testing.T) {
```
Comment (bogdan-rosianu) [inline update/container/accountDBSyncers_test.go:103]: should should

File: update/container/accountDBSyncers_test.go [hunk 6]
```diff
@@ -10,89 +10,169 @@ import (
...
+		t.Parallel()
+
+		adsc := NewAccountsDBSyncersContainer()
+		err := adsc.AddMultiple([]string{testKey0, testKey1, testKey1}, []update.AccountsDBSyncer{testAccountsDBSyncersVal0, testAccountsDBSyncersVal1, testAccountsDBSyncersVal1})
+		require.Equal(t, update.ErrContainerKeyAlreadyExists, err)
+	})
+	t.Run("should work"+
```
Comment (bogdan-rosianu) [inline update/container/accountDBSyncers_test.go:110]: remove `+""` ?

File: trie/trieStorageManagerInEpoch_test.go (hunk 4 above)
Comment (sstanculeanu) [reply trie/trieStorageManagerInEpoch_test.go:8]: decided to use errorsMx in a previous PR and kept the same everywhere

File: trie/snapshotTrieStorageManager_test.go (hunk 2 above)
Comment (sstanculeanu) [reply trie/snapshotTrieStorageManager_test.go:11]: decided to use errorsMx in a previous PR and kept the same everywhere

**Review by bogdan-rosianu — APPROVED** [review]

**Review by BeniaminDrasovean — APPROVED** [review]

**Pull request merged**